pylearn2/linear/cudnn2d.py

```diff
@@ -0,0 +1,259 @@
+                       kernel_shape,
+                       batch_size=None,
+                       subsample=(1, 1),
+                       border_mode = 'valid',
+                       message = "",
+                       rng = None):
+    """
+    .. todo::
+
+        WRITEME properly
+
+    Creates a CorrMM2D with random kernels
+
+    Parameters
+    ----------
+    irange : TODO
+    input_space : TODO
+    output_space : TODO
+    kernel_shape : 2D list or tuple
+    batch_size : int, optional
+    subsample : tuple, optional
+    border_mode : string, optional
+    message : string, optional
+    rng : optional
+    """
+
+    rng = make_np_rng(rng, default_seed, which_method='uniform')
+
+    W = sharedX(rng.uniform(-irange, irange, (output_space.num_channels,
+                                              input_space.num_channels,
+                                              kernel_shape[0],
+                                              kernel_shape[1])))
+
+    return Cudnn2D(
+        filters=W,
+        batch_size=batch_size,
+        input_space=input_space,
+        output_axes=output_space.axes,
+        subsample=tuple(subsample),
+        border_mode=border_mode,
+        filters_shape=W.get_value(borrow=True).shape,
+        message=message
+    )
+
+
+def make_sparse_random_conv2D(num_nonzero,
+                              input_space,
+                              output_space,
+                              kernel_shape,
+                              batch_size,
+                              subsample=(1, 1),
+                              border_mode='valid',
+                              message="",
+                              rng=None):
+    """
+    .. todo::
+
+        WRITEME properly
+
+    Creates a Cudnn2D with random kernels, where the randomly initialized
+    values are sparse
+    """
+
+    raise AssertionError(
+        "TODO: I think this is a bug--num_nonzero "
+        "determines the total number of nonzero elements in the "
+        "whole kernel stack, not the number of non-zero elements per "
+        "kernel. Investigate what it's meant to do."
+    )
+
+    rng = make_np_rng(rng, default_sparse_seed,
+                      which_method=['randn', 'randint'])
+
+    W = np.zeros((output_space.num_channels, input_space.num_channels,
+                  kernel_shape[0], kernel_shape[1]))
+
+    def random_coord():
+        return [rng.randint(dim) for dim in W.shape]
+
+    for i in xrange(num_nonzero):
+        o, ch, r, c = random_coord()
+        while W[o, ch, r, c] != 0:
+            o, ch, r, c = random_coord()
+        W[o, ch, r, c] = rng.randn()
+
+    W = sharedX(W)
+
+    return Cudnn2D(
+        filters=W,
+        batch_size=batch_size,
+        input_space=input_space,
+        output_axes=output_space.axes,
+        subsample=subsample,
+        border_mode=border_mode,
+        filters_shape=W.get_value(borrow=True).shape,
+        message=message
+    )
+
```

Comment:
Sorry, I am messing with my vim settings and it's not behaving as I expect. I will fix it.